### PR TITLE
host_policy singleton

### PR DESCRIPTION
**Pull request by mchernov-intel**

### CODE REVIEW

File: cpp/oneapi/dal/detail/policy.hpp [hunk 1]
```diff
@@ -43,7 +43,10 @@ class ONEDAL_EXPORT default_host_policy {};
 
 class ONEDAL_EXPORT host_policy : public base {
 public:
-    host_policy();
+    static host_policy& instance() {
```
Comment (rlnx) [inline cpp/oneapi/dal/detail/policy.hpp:46]: According to our conventions

```suggestion
    static host_policy& get_instance() {
```

**Review by rlnx — COMMENTED** [review]
I see changes only in `ops_dispatcher.hpp`. Seems like you missed a lot of entries:
image

File: cpp/oneapi/dal/detail/policy.hpp [hunk 2]
```diff
@@ -55,6 +58,8 @@ class ONEDAL_EXPORT host_policy : public base {
 private:
     void set_enabled_cpu_extensions_impl(const cpu_extension& extensions) noexcept;
 
+    host_policy();
```
Comment (rlnx) [inline cpp/oneapi/dal/detail/policy.hpp:61]: Keep this public, we still need to have multiple instances of policies

File: cpp/oneapi/dal/detail/policy.hpp (hunk 1 above)
Comment (rlnx) [reply cpp/oneapi/dal/detail/policy.hpp:46]: Does `static const host_policy&` work?

**Comment by mchernov-intel** [comment]
@orrrrtem please, provide approve if PR fixed performance degradation

**Comment by rlnx** [comment]
@orrrrtem, does it solve the problem with overhead caused by  `host_policy` constructor?

File: cpp/oneapi/dal/detail/policy.hpp [hunk 3]
```diff
@@ -45,6 +45,11 @@ class ONEDAL_EXPORT host_policy : public base {
 public:
     host_policy();
 
+    static host_policy& get_default() {
```
Comment (rlnx) [inline cpp/oneapi/dal/detail/policy.hpp:48]: Does `const` work here?

```suggestion
    static const host_policy& get_default() {
```

File: cpp/oneapi/dal/detail/policy.hpp [hunk 4]
```diff
@@ -45,6 +45,11 @@ class ONEDAL_EXPORT host_policy : public base {
...
+        static host_policy instance;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/detail/policy.hpp:49]:
```suggestion
        const static host_policy instance;
```

**Review by rlnx — APPROVED** [review]

**Comment by orrrrtem** [comment]
> @orrrrtem, does it solve the problem with overhead caused by `host_policy` constructor?

Sorry for the delay, I was on vacation. Let me take few hours to check.

**Review by orrrrtem — APPROVED** [review]
Vtune shows that the problem is resolved.
Thank you!!!

**Pull request merged**